pkg/function/overrides/override.go

```diff
@@ -0,0 +1,177 @@
+	if err != nil {
+		return nil, fmt.Errorf("failed to create program: %w", err)
+	}
+
+	//Value can be null which is abit wierd.
+	return p, nil
+
+}
+
+func (o *Override) Test(data map[string]interface{}) (bool, error) {
+	// Evaluate with the input data
+
+	prg, err := o.validate()
+	if err != nil {
+		return false, fmt.Errorf("failed to validate override: %w", err)
+	}
+
+	result, _, err := prg.Eval(data)
+	if err != nil {
+		return false, fmt.Errorf("failed to evaluate condition: %w", err)
+	}
+
+	// Convert result to boolean
+	if boolVal, ok := result.(types.Bool); ok {
+		return bool(boolVal), nil
+	}
+
+	return false, fmt.Errorf("condition did not evaluate to boolean, got: %T", result)
+}
+
+// AnnotateOverrides will take care of appropriatly serializng your overrides to annotations
+// merging them with others that exist
+func AnnotateOverrides(obj *unstructured.Unstructured, overrides []Override) error {
+
+	// Add Helm annotations that are required for Helm to recognize the resources
+	annotations := obj.GetAnnotations()
+	if annotations == nil {
+		annotations = make(map[string]string)
+	}
+	merged := overrides
+	if existingStr, exists := annotations["eno.azure.io/overrides"]; exists {
+		var existing []Override
+		json.Unmarshal([]byte(existingStr), &existing)
+		merged = append(merged, overrides...)
+	}
+
+	for _, override := range merged {
+		if _, err := override.validate(); err != nil {
+			return fmt.Errorf("validating override: %w", err)
+		}
+	}
+
+	jsonBytes, err := json.Marshal(merged)
+	if err != nil {
+		return fmt.Errorf("failed to marshal overrides: %w", err)
+	}
+
+	//should we append to existing annoations or panic if they exist?
+	annotations["eno.azure.io/overrides"] = string(jsonBytes)
+
+	obj.SetAnnotations(annotations)
+	return nil
+}
+
+// ReplaceIf uses overrides to create a conditonal eno.azure.io/replace that only applies when some condition is met
+// useful if you want server side apply most of the time except for some corner cases
+func ReplaceIf(condition string) (Override, error) {
+	true := "true"
+	o := Override{
+		Path:      `self.metadata.annotations["eno.azure.io/replace"]`,
+		Value:     &true,
+		Condition: condition,
+	}
+	//even if they didn't test ensure it valdiates
+	if _, err := o.validate(); err != nil {
+		return Override{}, fmt.Errorf("validating override: %w", err)
+	}
+	return o, nil
+}
+
+// AllowVPA lets VPA or external actor raise resources/requests for a given container (need to do limit too)
+func AllowVPA(container string, req corev1.ResourceRequirements) ([]Override, error) {
+	overrides := []Override{}
+	for rtype, value := range req.Requests {
+		if value.IsZero() {
+			continue // skip zero values
+		}
+		o, err := allowVPA(container, rtype.String(), "requests", value.String())
+		if err != nil {
+			return nil, fmt.Errorf("creating override for requests: %w", err)
+		}
+		overrides = append(overrides, o)
+	}
+	for rtype, value := range req.Limits {
```

Comment:
nit: could refactor into a helper func to avoid having two nearly identical loops